plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditor.java

```diff
@@ -846,8 +845,8 @@ public boolean isDirty() {
             return true;
         }
         if (isNonPersistentEditor()) {
-            // Console is never dirty
-            return false;
+            // Console is always dirty
+            return true;
```

Comment:
Won't it add * to the name of each console tab?

Reply:
It will. That's the goal.

Reply:
But doesn't it the main idea of console - to be unsaved editor? That's why we have them. And they have another icon to indicate it.

The * near the tab name shows unsaved state for editors, when the normal editor state is saved. But the normal console state is unsaved.

I don't like the idea of always having * for all console editors.

Reply:
It's up for a discussion.

Reply:
Perhaps we can override the editor's title to exclude the asterisk if it's a console.

Reply:
Unfortunately, it's impossible to override it without hacky constructions.

Keep it mind that it WON'T prompt the user to save itself upon closing.